Sage/carbon
# fix(filterable-select, multi-select): filter text fixes (FE-4163)

### Proposed behaviour
Input text typed by user in FilterableSelect and MultiSelect should not be cleared when option list is closed.
There should be an easy way to access filter text typed in by user, that could be achieved by adding the onFilterChange prop.

### Current behaviour
in FilterableSelect and MultiSelect closing the option list or replacing the option list clears user typed filter text.
This could be an issue when filterText change triggers a backend call - user will not be able to type the whole phrase. 

### Checklist


~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~

### Additional context
Fixes: #4155 

### Testing instructions
1. npm run
2. open http://localhost:9001/?path=/story/design-system-select-filterable--with-is-loading-prop
3. test if the text is cleared
4. open http://localhost:9001/?path=/story/design-system-select-filterable--default-story
5. check onFilterChange action
6. repeat for MultiSelect
